checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java

```diff
@@ -118,6 +119,12 @@ private void appendCustomFilters(Writer writer) throws IOException {
         writer.append(filtersXml);
     }
 
+    private void appendTabWidth(Writer writer) throws IOException {
+        final String tabWidth = configuration.get(CheckstyleConstants.CHECKER_TAB_WIDTH)
+                .orElse(null);
```

Comment:
I don't think we should default to `null`.  Use the new constant in `CheckstyleConstants`.

Reply:
Nevermind, if it is `null` the property isn't written to the writer. So it will use the default of checkstyle.

Reply:
It is better to avoid null passing around, can we skip printing property instead ?

Reply:
> can we skip printing property instead ?

This is essentially what `null` does.

Reply:
yes, but we avoid  null usage in plugin.
So plugin will not depend on this to the library.

Reply:
this line filters out the `null` or empty values:
https://github.com/checkstyle/sonar-checkstyle/blob/a0330ce67100f6d11b778ed5b53afc43c86e0176/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java#L228
I agree with @rnveach. Atm, I don't see a point in duplicating a null or empty check kinda one line before it's done anyways.

Reply:
ok, not a critical item